groceries_app/views.py

```diff
@@ -1,20 +1,41 @@
 from django.views.generic import TemplateView
-from groceries_app.models import Meal
+from django.views.generic.list import ListView
+from django.views.generic.detail import DetailView
+from groceries_app.models import Meal, IngredientQuantity
 
 
 class HomeView(TemplateView):
     template_name = 'home.html'
 
+
 class AboutView(TemplateView):
     template_name = 'about.html'
 
-class RecipesView(TemplateView):
-    template_name = 'recipes/recipes.html'
 
 class MealChoiceView(TemplateView):
     template_name = 'grocery_planner/gp1-meal-choice.html'
     
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['meal_list'] = Meal.objects.order_by('name')
+        return context
+
+
+class RecipeListView(ListView):
+    template_name = 'recipes/recipes_list.html'
+    model = Meal
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['meal_list'] = Meal.objects.order_by('name')
```

Comment:
`RecipeListView` and `MealChoiceView` are both displaying the meal_list?

Reply:
RecipeListView makes a query to get all of the meals, but will only display meals with instructions included. The MealChoiceView will show all of the meals.